server/controllers/admin/locations/getLocations.js

```diff
@@ -0,0 +1,15 @@
+const connectToDatabase = require("../../../database/dbConnection");
+const {
+  getAllRentalRecordsAdmin,
+} = require("../../../database/queries/rentalRecord");
+
+// admin can get all the records including info of the users who verified the data
+module.exports = async function getLocations(req, res) {
+  try {
+    await connectToDatabase();
```

Comment:
Since we are using express now and we are connecting to the db in the app.js I don't think this is required anymore
(this goes for the other controllers)

Reply:
Is there any way that the server might be up but the database connection would have closed? I feel like it's safer to keep this line in so that controllers can initialise a reconnection if they need to, but happy to take it out if you think that wouldn't be needed

Reply:
We are used to doing something like this but I don't know if this is the best idea. it's up to you to pick which seems better to you :)
```js
dbConnection().then(() => {
  app.listen(app.get("port"), () => {
    // eslint-disable-next-line no-console
    console.log("Listening on port: ", app.get("port"));
  });
}).catch((err) => {
  // handle failing db connection
  console.log("db connection failed", err.name);
});
```

Reply:
I think I would prefer to leave it as we have it, because then the server won't crash if the database disconnects, it will just start a new connection. If the database is already connected, then the function will return the cached connection so should not hurt performance